src/application.ts

```diff
@@ -40,10 +44,20 @@ export default class Application {
     this.processors = settings.processors || [];
     this.services = settings.services || ({} as ApplicationServices);
     this.defaultProcessor = settings.defaultProcessor || OperationProcessor;
-
+    this.addons = [];
     this.serializer = new (settings.serializer || JsonApiSerializer)();
   }
 
+  use(addon: typeof Addon, options?: AddonOptions) {
+    if (this.addons.find(installedAddon => installedAddon.addon === addon)) {
+      return;
+    }
+
+    new addon(this, options).install().then(() => {
```

Comment:
why is `then` used?

Reply:
@ebryn Because `app.use` is typically going to be used on a root scope and it would be unconfortable to use it as an async function, from a developer ergonomics standpoint.